setup.py

```diff
@@ -9,137 +8,80 @@ def main():
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','IR_Land','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','IR_Snow','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','IR_Water','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','MW_Water','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','VIS_Ice','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','VIS_Land','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','VIS_Snow','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-
-    p = os.path.join(cwd,'fix_crtm-internal_develop','EmisCoeff','VIS_Water','SEcategory','Little_Endian') 
-    for f in os.listdir(p):
-        shutil.copy(os.path.join(p,f), installLocation)
-def downloadAndMoveCoef(coef_path):
-    #if file is downloaded don't do it again.
-    if(not os.path.exists('fix_REL-2.4.0.tgz') ):
-        # if you're on discover, don't use ftp, because, well, you can't. Anyone else can talk to UCAR.
-        #if('discover' in socket.gethostname()):
-        url = 'https://gmao.gsfc.nasa.gov/gmaoftp/bkarpowi/crtm/fix_REL-2.4.0_le.tgz'
-        #else:
-        #    url = 'ftp://ftp.ucar.edu/pub/cpaess/bjohns/fix_REL-2.4.0.tgz'
-        downloadCoef(url)
-    extractCoef()
-    moveCrtmCoefficients(coef_path)
 
+    if( not os.path.isdir( coefDest )  ):
+        print("Creating directory:{}".format(coefDest))
+        os.makedirs( coefDest )
+    else:
+        print("Cleaning directory:{}".format(coefDest))
+        shutil.rmtree( coefDest )
+        print("Creating directory:{}".format(coefDest))
+        os.makedirs( coefDest )
+    print("Linking Coefficients in {} to {}".format(coefDir,coefDest))
+    topdir = os.listdir(coefDir)
+    # use os.walk to get files regardless if they're in an emc "fix" style directory, or crtm coeff tarball.
+    searchPath = coefDir 
+    toLink = []
+    filesPresent = []
+    for root,dirs,filez in os.walk(searchPath):
+        for name in filez:
+            srcPath = os.path.join(root,name)
+            if(not ('ODAS' in srcPath or 'Big_Endian' in srcPath)):
```

Comment:
@karpob as an FYI: future CRTMv3 after v3.2.x will be netCDF only.

Reply:
Yeah my thought is when it's netcdf only to move all of that out of setup.py into something like cartopy does for higher resolution maps where it downloads on request when it doesn't find anything. Also, at some point figure out how to remove setup.py completely.